scene/gui/line_edit.cpp

```diff
@@ -1913,15 +1913,15 @@ bool LineEdit::is_secret() const {
 }
 
 void LineEdit::set_secret_character(const String &p_string) {
-	// An empty string as the secret character would crash the engine.
-	// It also wouldn't make sense to use multiple characters as the secret character.
-	ERR_FAIL_COND_MSG(p_string.length() != 1, "Secret character must be exactly one character long (" + itos(p_string.length()) + " characters given).");
-
-	if (secret_character == p_string) {
+	String c = p_string;
+	if (c.length() > 1) {
+		WARN_PRINT("Secret character must be exactly one character long (" + itos(c.length()) + " characters given).");
+		c = c.left(1);
+	}
+	if (secret_character == c) {
```

Comment:
These changes affect https://github.com/godotengine/godot/issues/81681 and https://github.com/godotengine/godot/pull/81724.

The behavior difference should be documented in `LineEdit.xml`, as it's no longer an error to use a secret longer than 1 character (instead, it'll be trimmed to 1 character).

Also, are empty strings handled correctly?

Reply:
Empty strings are handled (it falls back to default value).